### PR TITLE
Few updates for api_key model

### PR DESCRIPTION
**Pull request by nghiant2710**
This PR contains the following changes:
```
- Add api_key.get_provisioning_api_keys_by_application method 
- Add api_key.get_device_api_keys_by_device method 
- Implement api_key.get_all_named_user_api_keys method
```

### CODE REVIEW

File: balena/auth.py [hunk 1]
```diff
@@ -12,6 +12,7 @@ class Auth(object):
     """
 
     _user_detail_cache = {}
+    _user_full_detail_cache = {}
```
Comment (thgreasi) [inline balena/auth.py:15]: I guess we should also clean this up from within `login`, `login_with_token` & `log_out`, like we do for `_user_detail_cache` , right?

**Review by thgreasi — COMMENTED** [review]
Other than that LGTM, feel free to self-certify

File: balena/auth.py (hunk 1 above)
Comment (nghiant2710) [reply balena/auth.py:15]: updated.

**Comment by nghiant2710** [comment]
@balena-ci I self-certify!

**Pull request merged**